client/src/views/Tests/TestForm/TestDetails.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { models } from '@yahalom-tests/common';
+import { Row, FormField, Select } from '../../../components';
+import { useAuth } from "../../../hooks";
+import { enumToArray, SwitchCamelCaseToHuman } from '../../../utils';
+
+const languages = enumToArray(models.enums.Language).map(SwitchCamelCaseToHuman);
+
+
+export const TestDetails: React.FC<models.dtos.TestDto> = (testProp) => {
+    const { activeStudyField, buildAuthRequestData } = useAuth();
+    //set test prop as component state.
+    const [test, setTest] = useState<models.dtos.TestDto>({ ...testProp });
```

Comment:
Please use an interface, and get the details with a Pick, as done in other files.
In addition, having the test as a state here is **very bad**, since we don't want this component to handle the test, but only call callbacks if needed.